extensions/amp-instagram/0.1/amp-instagram.js

```diff
@@ -120,6 +126,10 @@ class AmpInstagram extends AMP.BaseElement {
   layoutCallback() {
     const iframe = this.element.ownerDocument.createElement('iframe');
     this.iframe_ = iframe;
+
+    this.win.addEventListener(
+        'message', event => this.handleInstagramMessages_(event));
```

Comment:
Let's keep a reference to the `unlisten` function so we remove the listener in `unlayout` e.g.
(`listen` is a utility function that can be imported from `../../../src/event-helper`)
```
this.unlistenMessage_ = listen(this.win, 'message', this.handleInstagramMessages_.bind(this));
```
then in `unlayout`
```
if (this.unlistenMessage_) {
this.unlistenMessage_();
}
```